pkg/generators/openapi_test.go

```diff
@@ -1570,3 +1570,60 @@ map[string]interface{}{
 
 `, funcBuffer.String())
 }
+
+func TestEnum(t *testing.T) {
+	callErr, funcErr, assert, _, funcBuffer := testOpenAPITypeWriter(t, `
+package foo
+
+// EnumType is the enumType
```

Comment:
I was expecting this enum type comment to be included along with the enum value comments.

Reply:
The comment on the enum type, like the name of the enum type, is usually kind of implementation detail. For example.

```go
// Protocol defines network protocols supported for things like container ports.
type Protocol string
```

later
```go
// ServicePort represents the port on which the service is exposed
type ServicePort struct {
	// Optional if only one ServicePort is defined on this service: The
	// name of this port within the service.  This must be a DNS_LABEL.
	// All ports within a ServiceSpec must have unique names.  This maps to
	// the 'Name' field in EndpointPort objects.
	Name string

	// The IP protocol for this port.  Supports "TCP", "UDP", and "SCTP".
	Protocol Protocol
```
The comment on the field should be good enough.

Reply:
I agree that this is often not a particularly valuable comment.

But if a developer does add some really important information in this and we choose to omit it, and then as a result the API is misused, I don't think I could defend our decision to omit the the comment. I think we responsible for including everything author included in the godoc of the type in the openapi documentation.